### DISCUSSION TITLE
Orange Sky during daytime since update

### DISCUSSION THREAD

**Discussion by deydalus** [aaronwmorris/indi-allsky#1734] [Q&A] [answered]
Hello everybody and merry christmas,

I am running indi-allsky on a RPi5 with a vivotek fisheye fixed dome network camera. I gather the images with pyCurl from the camera. Since the update to indi_v2024.09.1 a couple days ago the colors are wrong. During the daytime I have a orange sky while the image on the camera is correct. With the former version (I'm not sure, maybe v2024.06 or something) it worked fine.

I changed the settings of the Bayern pattern but without any effect. I tried also some WB settings but wthout a solution. Now I'm running out of ideas and hope someone can help me.

Here are some pictures (directly from the camera and from indi):
![video](https://github.com/user-attachments/assets/fe4f1143-f9c4-40cd-8617-6375dd23f2af)
![ccd1_20241224_132214](https://github.com/user-attachments/assets/245ab1d3-d314-48cd-acc3-37585ad852f0)

BTW, I'm a newbie to indi and RPi ;-)

Best wishes
Holger

**Comment by aaronwmorris** [comment, marked as answer]
Merged #1737 to fix. 

The red and blue channels were reversed.  I definitely broke this a while ago.  I went back and reviewed the change where this broke and I do not know why I changed the code.  Thank you for opening this.

The history on this is OpenCV deals with BGR data, but the PIL library which I use to load images loads in RGB format, so the data has to be continually shuffled depending on what is handling the image.

**Reply by deydalus** [reply]
Wonderful, it works again. Thank you !!
